data_steward/cdr_cleaner/cleaning_rules/remove_invalid_procedure_source_records.py

```diff
@@ -57,11 +70,32 @@ def get_remove_invalid_procedure_source_queries(project_id, dataset_id):
     """
     queries_list = []
 
-    query = dict()
-    query[cdr_consts.
-          QUERY] = REMOVE_INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
-              dataset=dataset_id, project=project_id)
-    queries_list.append(query)
+    # queries to sandbox
+    invalid_records = dict()
+    invalid_records[
+        cdr_consts.QUERY] = INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
+            project=project_id,
+            dataset=dataset_id,
+            table=TABLE,
+            sandbox_dataset=get_sandbox_dataset_id(dataset_id),
+            intermediary_table=INTERMEDIARY_TABLE_NAME)
+    queries_list.append(invalid_records)
+
+    # queries to delete invalid procedure source records
+    valid_records = dict()
+    valid_records[
+        cdr_consts.QUERY] = VALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
+            project=project_id,
+            dataset=dataset_id,
+            table=TABLE,
+            sandbox_dataset=get_sandbox_dataset_id(dataset_id),
+            intermediary_table=INTERMEDIARY_TABLE_NAME)
+    queries_list.append({
+        clean_consts.QUERY: valid_records,
+        clean_consts.DESTINATION_TABLE: TABLE,
+        clean_consts.DESTINATION_DATASET: dataset_id,
+        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
+    })
```

Comment:
This creates this data structure:
```
{
query:       {'query': '\nSELECT * FROM `aou-res-curation-test.lrwb_deid_test_input_20190910.procedure_occurrence` as p0\nWHERE NOT EXISTS (\nSELECT p1.procedure_occurrence_id FROM `aou-res-curation-test.lrwb_deid_test_input_20190910_sandbox.procedure_occurrence_dc583` AS p1\nWHERE p0.procedure_occurrence_id = p1.procedure_occurrence_id)\n'}
destination_table_id:				procedure_occurrence
destination_dataset_id:				lrwb_deid_test_input_20190910
write_disposition:				WRITE_TRUNCATE
}
```

What we want to create is:
```
{
query: 
SELECT * FROM `aou-res-curation-test.lrwb_deid_test_input_20190910.procedure_occurrence` as p0
WHERE NOT EXISTS (
SELECT p1.procedure_occurrence_id FROM `aou-res-curation-test.lrwb_deid_test_input_20190910_sandbox.procedure_occurrence_dc583` AS p1
WHERE p0.procedure_occurrence_id = p1.procedure_occurrence_id)
destination_table_id:				procedure_occurrence
destination_dataset_id:				lrwb_deid_test_input_20190910
write_disposition:				WRITE_TRUNCATE
}
```

This is happening because `valid_records` is a dictionary.  You should add the `DESTINATION_TABLE`, `DESTINATION_DATASET`, and `DISPOSITION`  to the `valid_records` dictionary and then do `queries_list.append(valid_records)`

Reply:
FYI:  The actual query is correct.

Reply:
thank you! just saw this error when updating my unittest, will adjust both and push.

Reply:
great!

Reply:
fixed and pushed.